.github/workflows/ci-cd-workflow.yml

```diff
@@ -0,0 +1,248 @@
+        restore-keys: |
+          ${{ runner.os }}-notebooks-pip-
+
+    - uses: actions/cache@v1
+      if: startsWith(runner.os, 'Windows')
+      with:
+        path: ~\AppData\Local\pip\Cache
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/**/setup.py') }}
+        restore-keys: |
+          ${{ runner.os }}-notebooks-pip-
+
+    - name: Install notebook dependencies
+      run: |
+        pip install --upgrade pip
+        pip install -e .[tests,notebooks]
+    - name: Test notebooks with nbval
+      run: |
+          pytest notebooks -r a --nbval-lax --sanitize-with tests/notebook-tests.cfg --no-cov
+    - name: Test notebooks strictly
+      if: matrix.python-version == 3.7
+      run: |
+        pytest notebooks -r a --nbval --sanitize-with tests/notebook-tests.cfg --no-cov
+
+
+  deploy-test-pypi:
+    needs: test-notebooks
+    if: startsWith(github.ref, 'refs/tags/v')
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.7]
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Setup python
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        pip install --upgrade pip
+        pip install -e .[dev]
+    - name: Create package
+      run: python setup.py sdist bdist_wheel --universal
+    - name: Publish package to TestPyPI
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        user: __token__
+        password: ${{ secrets.test_pypi_password }}
+        repository_url: https://test.pypi.org/legacy/
+
+  test-pypi-install:
+    needs: deploy-test-pypi
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.7]
+
+    steps:
+    - name: Setup python
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install package from test PyPI
+      run: |
+        pip install --upgrade pip
+        pip install cftime openscm_units pandas pint xarray
+        pip install -i https://testpypi.python.org/pypi pymagicc --no-dependencies --pre
+    - name: Test installation
+      run: |
+        python -c "import sys; sys.path.remove(''); import pymagicc; print(pymagicc.__version__)"
+
+  deploy-pypi:
+    needs: test-pypi-install
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.7]
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Setup python
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        pip install --upgrade pip
+        pip install -e .[dev]
+    - name: Create package
+      run: python setup.py sdist bdist_wheel --universal
+    - name: Publish package to PyPI
+      uses: pypa/gh-action-pypi-publish@master
```

Comment:
> Pro tip: instead of using branch pointers, like master, pin versions of Actions that you use to tagged versions or sha1 commit identifiers. This will make your workflows more secure and better reproducible, saving you from sudden and unpleasant surprises.
https://github.com/pypa/gh-action-pypi-publish

I'd pin that to a sha hash

Reply:
```suggestion
      uses: pypa/gh-action-pypi-publish@v1
```

?? How do you mean?